tinkerers-ci/scripts/refresh-token.sh

```diff
@@ -1,10 +1,86 @@
 #!/usr/bin/env bash
 
-GITHUB_TOKEN=$( /scripts/generate_token.sh )
+#-------------------------
+# Environment variables that need to be set
+#-------------------------
+
+# github app id
+#app_id=284804
+
+# github app installation id
+#installation_id=33432247
+
+# pem file generated for github-app
+# app_private_key=""
+
+#-------------------------
+# Generate JWT using pem 
+#-------------------------
+
+header='{
+    "alg": "RS256",
+    "typ": "JWT"
+}'
+payload_template='{}'
+
+build_payload() {
+        # shellcheck disable=SC2154
+        jq -c \
+                --arg iat_str "$(date +%s)" \
+                --arg app_id "${app_id}" \
+        '
+        ($iat_str | tonumber) as $iat
+        | .iat = $iat
+        | .exp = ($iat + 300)
+        | .iss = ($app_id | tonumber)
+        ' <<< "${payload_template}" | tr -d '\n'
+}
+
+b64enc() { openssl enc -base64 -A | tr '+/' '-_' | tr -d '='; }
+json() { jq -c . | LC_CTYPE=C tr -d '\n'; }
+rs256_sign() { openssl dgst -binary -sha256 -sign <(printf '%s\n' "$1"); }
+
+algo=${1:-RS256}; algo=${algo^^}
```

Comment:
I wasn't aware of this so might be nice to have a comment :) 

```suggestion
algo=${1:-RS256}; 
algo=${algo^^} # Convert to uppercase 
```